gateway-ha/src/test/java/io/trino/gateway/ha/router/TestResourceGroupsManager.java

```diff
@@ -50,27 +50,35 @@ public void testCreateResourceGroup()
     {
         ResourceGroupsDetail resourceGroup = new ResourceGroupsDetail();
 
-        resourceGroup.setResourceGroupId(0L);
+        resourceGroup.setResourceGroupId(0L);   // This should be ignored. Id are generated by database (AUTO_INCREMENT) starting from 1.
         resourceGroup.setName("admin");
         resourceGroup.setHardConcurrencyLimit(20);
         resourceGroup.setMaxQueued(200);
         resourceGroup.setJmxExport(true);
         resourceGroup.setSoftMemoryLimit("80%");
-
-        ResourceGroupsDetail newResourceGroup = resourceGroupManager.createResourceGroup(resourceGroup,
+        ResourceGroupsDetail adminResourceGroup = resourceGroupManager.createResourceGroup(resourceGroup,
                 null);
+        assertThat(adminResourceGroup).isEqualTo(resourceGroup);
 
-        assertThat(newResourceGroup).isEqualTo(resourceGroup);
+        resourceGroup.setResourceGroupId(0L);   // Id are generated by database (AUTO_INCREMENT). This will be 2.
```

Comment:
would it make more sense to NOT have a setter at all? Or is this something that JDBI doesnt like

Reply:
Or do we want another ID that we generate (like a UUID string or so?)

Reply:
I agree that's confusing. I'll delete these two lines.
This setter will remain for Jackson deserialization.